components/script/dom/cssstyledeclaration.rs

```diff
@@ -400,7 +424,8 @@ impl CSSStyleDeclarationMethods for CSSStyleDeclaration {
 
     // https://dev.w3.org/csswg/cssom/#the-cssstyledeclaration-interface
     fn IndexedGetter(&self, index: u32) -> Option<DOMString> {
-        self.owner.with_block(|pdb| {
+        self.owner.with_mut_block(|pdb| {
+            pdb.deduplicate();
```

Comment:
Repeated calls will only deduplicate once and set a boolean flag to signal that deduplicating again is not needed. Some operations like adding a new declaration reset the flag.